### DISCUSSION TITLE
How to update Nion Swift on microscope user-PC (window OS)

### DISCUSSION THREAD

**Discussion by PhysXu** [nion-software/nionswift#584] [General]
Hi, Chris,
Is there any instruction with details for updating our swift software on the microscope user-PC (window OS) to the latest version?
We also want to retain the old version at the same time, in case there are some bugs in the new one.

Thanks!

**Comment by cmeyer** [comment]
We tend to roll out new major releases (0.15) slowly to Nion microscope users so that we understand what issues arise. The microscope installations involve software that isn't open source and public, and often involve updates to non-Swift software such as microscope control and camera support. Unless otherwise instructed, it is usually best for Nion support to make the upgrades.

For offline use (not directly running a microscope), updates are much less risky. I am expanding the [documentation for offline use](https://nionswift.readthedocs.io/en/latest/index.html) with each minor release.

Can you email me directly to set up an update installation? (I'm not posting my email since this is a public forum - but I'm sure you have it).

**Reply by PhysXu** [reply]
Thanks for your reply.
I will contact you by email later.
